### PR TITLE
iOS: Dismiss Toasts and SnackBars on swipe or tap

### LINKED ISSUE #1225: [Enhancement] Swipe-to-dismiss for notifications
## Summary

Hi.

Currently there is no way for a user to prematurely dismiss SnackBar notifications. This is a very important feature as users are conditioned to swipe away lingering notifications, and it's impossible to select a time to display that is long enough for the users to read the contents but short enough that the notification won't become bothersome.

Furthermore, no matter how well the position of the notification has been designed it's bound to obscure some form of UI navigation.

Hence I think that swipe-to-dismiss functionality would be an essential enhancement. If it's hard to implement in the meantime I would be completely fine with a small 'X' button in the upper right corner that users could press, or other such mechanism.

Thank you!

### PR DESCRIPTION
**Pull request by humblehacker**
### Description of Change ###

On iOS, dismiss Toasts and SnackBars on swipe or tap. Bare minimum necessary to allow user to dismiss Toast/SnackBar before timeout expires. 

No tests added as there does not appear to be any UI tests in the project and adding them is non-trivial.

### Bugs Fixed ###

- Fixes #1225

### API Changes ###

None.

### Behavioral Changes ###

On iOS only, Toasts/Snackbars will be dismissed when tapped or swiped in any direction. Android behavior is unchanged.

### PR Checklist ###

- [ ] Has a linked Issue, and the Issue has been `approved`
    - Has linked issue, but not yet approved.
- [ ] Has tests (if omitted, state reason in description)
- [x] Has samples (if omitted, state reason in description)
    - Samples pre-existing
- [x] Rebased on top of main at time of PR
- [x] Changes adhere to coding standard

- [ ] Updated [documentation](https://github.com/MicrosoftDocs/xamarin-communitytoolkit)

### CODE REVIEW

**Comment by pictos** [comment]
@VladislavAntonyuk do you mind reviewing this PR?

File: src/CommunityToolkit/Xamarin.CommunityToolkit/Views/Snackbar/Helpers/NativeSnackBar.ios.macos.cs [hunk 1]
```diff
@@ -99,9 +100,45 @@ public NativeSnackBar Show()
...
+		{
+#if __IOS__
+			if (SnackBarView == null)
+				return;
+
+			var tapGesture = new UITapGestureRecognizer();
+			tapGesture.AddTarget(OnTap);
+			SnackBarView.AddGestureRecognizer(tapGesture);
+
+			var horizontalSwipeGesture = new UISwipeGestureRecognizer();
```
Comment (VladislavAntonyuk) [inline src/CommunityToolkit/Xamarin.CommunityToolkit/Views/Snackbar/Helpers/NativeSnackBar.ios.macos.cs:118]: Why do we need 2 gesture recognizes?
Can we combine them into one?

File: src/CommunityToolkit/Xamarin.CommunityToolkit/Views/Snackbar/Helpers/NativeSnackBar.ios.macos.cs [hunk 2]
```diff
@@ -99,9 +100,45 @@ public NativeSnackBar Show()
...
+			SnackBarView.AddGestureRecognizer(horizontalSwipeGesture);
+
+			var verticalSwipeGesture = new UISwipeGestureRecognizer();
+			verticalSwipeGesture.Direction = UISwipeGestureRecognizerDirection.Down | UISwipeGestureRecognizerDirection.Up;
+			verticalSwipeGesture.AddTarget(OnSwipe);
+			SnackBarView.AddGestureRecognizer(verticalSwipeGesture);
+#endif
+		}
+
+		private void OnTap()
```
Comment (VladislavAntonyuk) [inline src/CommunityToolkit/Xamarin.CommunityToolkit/Views/Snackbar/Helpers/NativeSnackBar.ios.macos.cs:130]: This code has no references on macOS.
Why is it only for iOS? Can we add the same for macOS?

File: src/CommunityToolkit/Xamarin.CommunityToolkit/Views/Snackbar/Helpers/NativeSnackBar.ios.macos.cs [hunk 3]
```diff
@@ -99,9 +100,45 @@ public NativeSnackBar Show()
...
 		}
 
+		private void AddDismissGestures()
+		{
+#if __IOS__
+			if (SnackBarView == null)
+				return;
+
+			var tapGesture = new UITapGestureRecognizer();
+			tapGesture.AddTarget(OnTap);
```
Comment (VladislavAntonyuk) [inline src/CommunityToolkit/Xamarin.CommunityToolkit/Views/Snackbar/Helpers/NativeSnackBar.ios.macos.cs:115]: Do we need TapGesture?
I think the original issue is about swipe to dismiss.

**Comment by VladislavAntonyuk** [comment]
I will be able to run and test it on Monday.
I left some comments.
Maybe we should add a bindable property to allow/disallow this functionality.

**Comment by brminnick** [comment]
Thanks! However, we are no longer adding new features to Xamarin Community Toolkit, focusing on the [.NET MAUI Community Toolkit](https://github.com/communitytoolkit/maui).

Please open a [New Feature Discussion](https://github.com/CommunityToolkit/Maui/discussions/new) to implement this feature in the .NET MAUI Community Toolkit.

I've posted more information about the Future Of Xamarin Community Toolkit here: https://devblogs.microsoft.com/xamarin/the-future-of-xamarin-community-toolkit/?WT.mc_id=mobile-0000-bramin

**Pull request closed without merging**